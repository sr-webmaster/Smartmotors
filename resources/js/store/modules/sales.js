import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

const MODULE_URL = '/sales/v1'
export const _getItemPath = ({ id }) => `${MODULE_URL}/sales/${id}`

export default ({ items, item }, options) => {
  const store = makeRestStore('sales', { items, item }, {
    itemsPath: () => `${MODULE_URL}/sales`,
    itemPath: _getItemPath
  }, options)

  return {
    namespaced: true,
    modules: {
      export: makeRestStore(
        'export',
        { items, item },
        {
          itemsPath: () => `${MODULE_URL}/export`,
          itemPath: () => `${MODULE_URL}/export`
        }
      )
    },
    ...store
  }
}
