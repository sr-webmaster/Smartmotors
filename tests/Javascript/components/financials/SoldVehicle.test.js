import { mount } from '@vue/test-utils'
import Component from '~/components/financials/SoldVehicle.vue'
import { FIXTURE_ELASTICSEARCH_DEAL } from 'tests/__data__/sales'
import { FIXTURE_ELASTICSEARCH_VEHICLES } from 'tests/__data__/vehicles'

describe('SoldVehicle', () => {
  describe('Snapshot', () => {
    test('default', async () => {
      const wrapper = mount(Component)
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('with mocked data', async () => {
      const wrapper = mount(Component, {
        propsData: {
          pbsData: FIXTURE_ELASTICSEARCH_DEAL.Vehicles[0],
          vehicles: FIXTURE_ELASTICSEARCH_VEHICLES
        }
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.find('h3').text()).toEqual('NEW TOYOTA RAV4 2018')
      expect(wrapper.find('.v-chip').text()).toContain('M464621')
      expect(wrapper.find('.f-soldVehicle_VIN input').element.value).toEqual('JTMBFREV6HJ116034')
      expect(wrapper.find('.f-soldVehicle_mileage input').element.value).toEqual('0')
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Computed', () => {
    test('type used', async () => {
      const wrapper = mount(Component, {
        propsData: {
          pbsData: FIXTURE_ELASTICSEARCH_DEAL.Vehicles[1]
        }
      })
      expect(wrapper.vm.type).toEqual('USED')
    })
  })
})
