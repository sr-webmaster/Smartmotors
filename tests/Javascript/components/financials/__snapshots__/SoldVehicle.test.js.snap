// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoldVehicle Snapshot default 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__text"
  >
    <div
      class="container grid-list-md"
    >
      <div
        class="layout column"
      >
        <div
          class="flex"
        >
          <h3
            class="primary--text"
          >
            
               
          
          </h3>
           
          <span
            class="v-chip v-chip--small theme--light info white--text"
            tabindex="0"
          >
            <span
              class="v-chip__content"
            >
              
            Stock #: 
          
            </span>
          </span>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-soldVehicle_VIN v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    VIN #
                  </label>
                  <input
                    aria-label="VIN #"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-soldVehicle_daysInStock v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Days in stock
                  </label>
                  <input
                    aria-label="Days in stock"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-soldVehicle_mileage v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Mileage
                  </label>
                  <input
                    aria-label="Mileage"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SoldVehicle Snapshot with mocked data 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__text"
  >
    <div
      class="container grid-list-md"
    >
      <div
        class="layout column"
      >
        <div
          class="flex"
        >
          <h3
            class="primary--text"
          >
            
            NEW TOYOTA RAV4 2018
          
          </h3>
           
          <span
            class="v-chip v-chip--small theme--light info white--text"
            tabindex="0"
          >
            <span
              class="v-chip__content"
            >
              
            Stock #: M464621
          
            </span>
          </span>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-soldVehicle_VIN v-text-field v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    VIN #
                  </label>
                  <input
                    aria-label="VIN #"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-soldVehicle_daysInStock v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Days in stock
                  </label>
                  <input
                    aria-label="Days in stock"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-soldVehicle_mileage v-text-field v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Mileage
                  </label>
                  <input
                    aria-label="Mileage"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
